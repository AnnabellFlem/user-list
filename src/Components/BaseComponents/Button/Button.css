@import '../../../Styles/colors.css';
@import '../../../Styles/sizes.css';

.btn {
  font-weight: 600;
  background-color: var(--orange-100);
  border-radius: var(--border-radius);
  color: var(--white);
  width: 296px;
  display: inline-flex;
  justify-content: center;
  cursor: pointer;
  transition: background-color 0.15s, color 0.15s;
  align-items: center;
  border: none;
}

.btn:hover {
  background-color: var(--orange-80);
}

.btn:active {
  background-color: var(--orange-120);
}

.btn:focus {
  outline: none;
  box-shadow: 4px 0 0 var(--orange-120),
  0 4px 0 var(--orange-120),
  -4px 0 0 var(--orange-120),
  0 -4px 0 var(--orange-120);
}

.btn--fullWidth {
  width: 100%;
}

.btn--s {
  font-size: var(--font-xs);
  line-height: var(--lh-s);
  padding: 14px 24px;
}

.btn--m {
  font-size: var(--font-s);
  line-height: var(--lh-m);
  padding: 20px 24px;
}
