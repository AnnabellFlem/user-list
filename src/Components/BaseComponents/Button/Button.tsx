import React from 'react'
import classNames from 'classnames'
import './Button.css'

type Props = React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
> &
  Partial<{
    size: 's' | 'm'
    fullWidth: boolean
  }>

const Button: React.FC<Props> = ({
  size = 'm',
  fullWidth,
  children,
  ...rest
}) => {
  const defaultButtonClassName = 'btn'
  const buttonClassNames = classNames(defaultButtonClassName, {
    [`${defaultButtonClassName}--${size}`]: size,
    [`${defaultButtonClassName}--fullWidth`]: fullWidth,
  })

  return (
    <button className={buttonClassNames} {...rest}>
      {children}
    </button>
  )
}

export default Button
