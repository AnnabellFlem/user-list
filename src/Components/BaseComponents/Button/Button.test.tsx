import React from 'react'
import { shallow } from 'enzyme'
import Button from './index'

describe('MoneyCell test', function () {
  let wrapper: any

  beforeAll(() => {
    wrapper = shallow(<Button />)
  })

  it('should to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render button', () => {
    expect(wrapper.find('button')).toBeDefined()
    expect(wrapper.find('button')).toHaveLength(1)
  })

  it('Should have an m size by default', () => {
    expect(wrapper.find('.btn--m')).toHaveLength(1)
  })

  it('Should have fullWidth prop', () => {
    wrapper = shallow(<Button fullWidth />)
    expect(wrapper.find('.btn--fullWidth')).toHaveLength(1)
  })
})
