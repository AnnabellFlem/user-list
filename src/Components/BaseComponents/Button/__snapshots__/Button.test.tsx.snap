// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyCell test should to match snapshot 1`] = `ShallowWrapper {}`;
